src/dotnet/commands/dotnet-build/Program.cs

```diff
@@ -93,10 +105,55 @@ public static int Run(string[] args)
 
                 msbuildArgs.AddRange(app.RemainingArguments);
 
-                return new MSBuildForwardingApp(msbuildArgs).Execute();
+                ret._forwardingApp = new MSBuildForwardingApp(msbuildArgs, msbuildPath);
+
+                return 0;
             });
 
-            return app.Execute(args);
+            int exitCode = app.Execute(args);
+            if (ret._forwardingApp == null)
+            {
+                throw new CodeNotExecutedException(exitCode);
+            }
+
+            return ret;
+        }
+
+        public static int Run(string[] args)
+        {
+            DebugHelper.HandleDebugSwitch(ref args);
+
+            BuildCommand cmd;
+            try
+            {
+                cmd = FromArgs(args);
+            }
+            catch (CodeNotExecutedException e)
+            {
+                return e.ExitCode;
+            }
+
+            return cmd.Execute();
+        }
+
+        public ProcessStartInfo GetProcessStartInfo()
+        {
+            return _forwardingApp.GetProcessStartInfo();
```

Comment:
Currently there are two ways we construct this object:
- public constructor which always creates _forwardingApp
- FromArgs which throws exception if this is null - @jgoshi this is another reason to throw exception instead of null check

While we could make this check here I think it simplifies a lot if we can have some assumptions about non-null fields - that's mostly why I prefer to throw and not do any null checks